static/app/components/draggableTabs/draggableTabList.tsx

```diff
@@ -71,7 +71,7 @@ function BaseDraggableTabList({
   useEffect(() => {
     setTabListState(state);
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [state.selectedItem, state.selectedKey, props.children]);
+  }, [state.selectedKey]);
```

Comment:
This is what was causing infinite rerenders when attempting to use `useContext(TabsContext)`. `state.selectedItem` is the content inside the tab, and since there are three dynamic elements in each tab, this is likely what was triggering the useEffect over and over again. For our use case, we only need to update the state when the selected key has changed.